database/factories/ModelFactory.php

```diff
@@ -98,6 +99,31 @@
 });
 
 /**
+ * Trader with linked market and sponsor.
+ */
+$factory->defineAs(App\Trader::class, 'with_market_sponsor', function ($faker) use ($factory) {
+    $trader = $factory->raw(App\Trader::class);
+
+    // Setup Sponsor.
+    $sponsor_def = $factory->raw(App\Sponsor::class);
+    $sponsor_def = array_merge($sponsor_def, [
+        'id' => 100,
+    ]);
+    factory(App\Sponsor::class)->create($sponsor_def);
+
+    // Setup Market.
+    $market_def = $factory->raw(App\Market::class);
+    $market_def = array_merge($market_def, [
+        'sponsor_id' => 100,
```

Comment:
I don't know - feels arbitrary to use 100. It should be ok to just use the 'next available id' and grab it with $market_def->id later.

Usually we use these specific factories to make things with fixed values on some attributes - like has a value vs. null. Or has a specific value. Like Trader from Mars factory(Trader::class, trader_from_mars)->create();

I may have led you down the garden path on this one... if this factory is only useful for your test right now AND we are making the relationship required later - we should probably ditch this. The array merge I meant just for the Trader... so it's getting all complicated.